src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,214 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
 #endif
 }
 
+TEST_CASE("LegerTxnRoot prefetch soroban entries", "[ledgertxn]")
+{
+    Config cfg = getTestConfig();
+    cfg.ENTRY_CACHE_SIZE = 10;
+
+    // Test setup.
+    VirtualClock clock;
+    cfg.EXPERIMENTAL_BUCKETLIST_DB = true;
+    Application::pointer app = createTestApplication(clock, cfg);
+    UnorderedSet<LedgerKey> keysToPrefetch;
+    auto& root = app->getLedgerTxnRoot();
+    LedgerTxn ltx(root);
+
+    auto lkMeterCold = std::make_shared<LedgerKeyMeter>();
```

Comment:
I don't quite get why do we have two meters here; in reality we always have just a single meter, right? It seems like you just need a separate scenario for the case covered with `lkMeterHot` and have just a single meter.

Reply:
This was to ensure that there is not "residual" quota / bad state from the previous prefetch call. In `LedgerManagerImpl::prefetchTransactionData` we create a new `LedgerKeyMeter` for the call to `prefetch`, it doesn't persist between prefetches.

Reply:
I see, I haven't realized that it's used in the closure. I think for the check you rather need unlimited quota, right? So could you just pass nullptr?

Reply:
If we make the "hot" prefetch call with unlimited quota, then the failure case ("don't load entries without quota") will succeed.

Reply:
Hmm, I feel like this could somehow be made more clear (maybe via separating the success/failure tests more), but I don't think that should block this PR, the current test seems fine.